### PR TITLE
charliecloud: init at 0.2.2

### PR DESCRIPTION
**Pull request by bzizou**
###### Motivation for this change
Want to use charliecloud :-)

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/virtualization/charliecloud/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{ stdenv, fetchurl, bash, python }:
+
+stdenv.mkDerivation rec {
+
+  version = "0.2.2";
+  name = "charliecloud-${version}";
+  bats_version = "0.4.0";
+
+ srcs = 
+    [ (fetchurl {
```
Comment (nlewo) [inline pkgs/applications/virtualization/charliecloud/default.nix]: Please use fetchFromGithub

File: pkgs/applications/virtualization/charliecloud/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+         url = "https://github.com/hpc/charliecloud/archive/v${version}.tar.gz";
+         sha256 = "0djq3j6q7zj4ignf1b87k64bg4h5aw8l0mxnshagsaskwvmhakpi";
+       })
+      (fetchurl {
+         url = "https://github.com/sstephenson/bats/archive/v${bats_version}.tar.gz";
```
Comment (nlewo) [inline pkgs/applications/virtualization/charliecloud/default.nix]: `bats` is already available in nixpkgs. Would it be possible to use?

File: pkgs/applications/virtualization/charliecloud/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+         sha256 = "1myqq56kzwqc7p3inxiv2wgc06kfy3rjf980s5wfw7k8y5j8s3a8";
+       })
+    ];
+
+  patches = [ ./CONDUCT.md_not_present_into_bats_release.patch ];
+
+  buildInputs = [ bash python ];
```
Comment (nlewo) [inline pkgs/applications/virtualization/charliecloud/default.nix]: I guess `bash` is provided by stdenv so, it is not necessary to include it.

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -14479,6 +14479,8 @@ with pkgs;
     amdappsdk = amdappsdk28;
   };
 
+  charliecloud = callPackage ../applications/virtualization/charliecloud { };
+
```
Comment (nlewo) [inline pkgs/top-level/all-packages.nix]: Too much spaces

**Review by nlewo — CHANGES_REQUESTED** [review]

**Comment by bzizou** [comment]
Charliecloud has a lot of new versions and changes. I'll do another PR with the latest.

**Pull request closed without merging**